src/architecture.c

```diff
@@ -322,37 +327,57 @@ static void arithm(arch_core* core)
 
 static void mov_imm(arch_core* core)
 {
-
+	arch_uint eff_data = ((cond_imm_data)core->regs.ir.data).addr;
+  	arch_uint eff_reg = ((cond_imm_data)core->regs.ir.data).breg; // Or use dreg??
+  	*help_get_reg(core, eff_reg) = eff_data;
 }
 
 static void add_imm(arch_core* core)
 {
-	
+	arch_uint eff_data = ((arith_data)core->regs.ir.data).src1 + ((arith_data)core->regs.ir.data).src2;
+  	arch_uint eff_reg = ((cond_imm_data)core->regs.ir.data).breg;
```

Comment:
All the immediate instructions use the conditional branch format, not the arithmetic format. you would not have src1 or src2 available.

You would do this instead:
```c
arch_uint_eff_data = ((cond_imm_data)core->regs.ir.data).addr;
arch_uint eff_reg = ((cond_imm_data)core->regs.ir.data).breg;
*help_get_reg(core, eff_reg) += eff_data;
```
Make your other immediate instructions almost identical to this.

Remember, the d-reg is equal to zero when the `address` field represents an immediate value. Even if you could, you wouldn't want to treat b-reg and d-reg as src1 and src2 in this case.

The wiki page for the instructions will be done sometime today, so look at that in about an hour or so to help you better understand.